modules/sitemaps/sitemaps.php

```diff
@@ -387,9 +387,21 @@ private function schedule_sitemap_generation() {
 		);
 
 		if ( ! wp_next_scheduled( 'jp_sitemap_cron_hook' ) ) {
-			wp_schedule_single_event( time() + 60, 'jp_sitemap_cron_hook' );
+			/**
+			 * Filter maximum number of minutes to randomly wait until generating sitemaps. Default 15.
+			 *
+			 * This filter allows a site operator or hosting provider to potentialy spread out sitemap generation for a
+			 * lot of sites over time. By default, it will be randomly done over 15 minutes.
+			 *
+			 * @module sitemaps
+			 * @since 6.6.1
+			 *
+			 * @param int $maximum Maximum number of minutes.
+			 */
+			$randomness = MINUTE_IN_SECONDS * mt_rand( 1, apply_filters( 'jetpack_sitemap_generation_randomness', 15 ) ); // Randomly space it out to start within next fifteen minutes.
```

Comment:
I wonder if it might be more useful to filter the whole delay:

```
$delay = apply_filters( 'jetpack_sitemap_generation_randomness', MINUTE_IN_SECONDS * mt_rand( 1, 15 ) ); // Randomly space it out to start within next fifteen minutes.

// later schedule for time() + $delay
```

Reply:
I don't have a preference. Mine is a bit more opinionated, but I suppose a hosting provider may want to delay it by some other factor based on their internal systems ("site id" or the db cluster it is on, etc etc). Updating.